src/commands/mod.rs

```diff
@@ -535,6 +537,82 @@ struct NewArgs {
     insert_before: bool,
 }
 
+/// Move the current working copy commit to the next child revision in the
+/// repository.
+///
+///
+/// The command moves you to the next child in a linear fashion.
+///
+///
+/// F        F  @
+/// |        | /
+/// C  @ =>  C
+/// | /      |
+/// B        B
+///
+///
+/// If `edit` is passed as an argument, it will move you directly to the child
+/// revision.
+///
+///
+/// F        F
+/// |        |
+/// C        C
+/// |        |
+/// B  @  => @
+/// | /      |
+/// A        A
```

Comment:
We'd move to C. `jj next` is as discussed just `jj new @-+ & ~@` and with --edit `jj edit @-+ & ~@`. 
Is the drawing unclear? I drew the working-copy commit as attached to its parent, as it matches my understanding from our previous conversation.

Reply:
No problem with the drawing, I'm just trying to understand how it will behave.

How about this case?
```
C   E
|     |
@  D
| /
A
```

If C had not existed, we said that `jj next` would create a new working copy on top of D. What would `jj next --edit` do? I think you're saying that it would edit commit D, but I'm wondering if it should instead edit commit C, and if C doesn't exist, it will error out. Maybe someone who uses this workflow can help? Maybe @talpr?

Reply:
> If C had not existed, we said that `jj next` would create a new working copy on top of D. What would `jj next --edit` do? I think you're saying that it would edit commit D

Yes, as it is consistent behavior. Looking at the graph though, I think it should edit E, as it is the logical child.


> but I'm wondering if it should instead edit commit C, and if C doesn't exist, it will error out.

Did you mean commit E?
 
As currently implemented, `jj next --edit` will only work if there is an immediate child and will error out if it doesn't exist.

Reply:
That's also my question. If `--edit` enforces "editing" workflow (i.e. the current position in stack is `@`, not `@-`), `jj next --edit` would be equivalent to `jj edit @+`. Otherwise repeated `jj next --edit` would be different than `jj next --edit N`.

Reply:
Looking back at our conversation from January, starting with a similar question [here][start]. Back then my reply was that it should "just" do a linear movement, not expecting it to move off branches ([reply][a1]). 

After that we had a back and forth, on where the actual working-copy commit should be created. 
Then I said that my expectation for `jj next/prev` were similar to `edit @- & ~@` and `edit @-+ & ~@` and yours rather to `new @- & @~` and `new @- & ~@`. The final point was disallowing `next --edit` on head commits, as it doesn't make sense, which we both agreed upon ([last reply][last]). 

I think my initial position is still correct, that we shouldn't move off branches. But other things still are in limbo.

I hope this summary helps everyone to form a clear model, what `next` and `prev` should and shouldn't do.  

[start]: https://discord.com/channels/968932220549103686/969291218347524238/1068950662852907009
[a1]: https://discord.com/channels/968932220549103686/969291218347524238/1069326922208788511
[last]: https://discord.com/channels/968932220549103686/969291218347524238/1069702727162151014